stdlib/public/core/StringCreate.swift

```diff
@@ -15,12 +15,30 @@
 internal func _allASCII(_ input: UnsafeBufferPointer<UInt8>) -> Bool {
   // NOTE: Avoiding for-in syntax to avoid bounds checks
   //
-  // TODO(String performance): Vectorize and/or incorporate into validity
-  // checking, perhaps both.
+  // TODO(String performance): SIMD-ize
   //
   let ptr = input.baseAddress._unsafelyUnwrappedUnchecked
   var i = 0
-  while i < input.count {
+
+  let count = input.count
+  let stride = MemoryLayout<UInt>.stride
+  let address = Int(bitPattern: ptr)
+
+  while (address &+ i) % stride != 0 && i < count {
+    guard ptr[i] <= 0x7F else { return false }
+    i &+= 1
+  }
+
+  while (i &+ stride) <= count {
+    let bytes: UInt = UnsafePointer(
+      bitPattern: address &+ i
+    )._unsafelyUnwrappedUnchecked.pointee
+    guard bytes & UInt(truncatingIfNeeded: 0x8080_8080_8080_8080 as UInt64)
```

Comment:
We should give the compiler folks a bug to unroll this so we don't have to do it by hand ;)

(Also, does truncatingIfNeeded elide truncation checks when given a constant like this? We definitely don't want any unnecessary branches in this loop)

Reply:
Do you mean do the whole operation, or just the promoting to a higher stride? CC @eeckstein @aschwaighofer 

(As wacky as it sounds, `truncatingIfNeeded` is the way to skip checks)

Reply:
Both would be nice! But I meant doing multiple words per loop, which this doesn't currently do. We squeezed another 1.5x out of CF's by doing that.

Reply:
BTW, the first loop is unrolled AFAICT

Reply:
Nice. That matches my mental model of the state of LLVM's loop unroller, which is that it handles known trip counts well, but mostly bails on unknown trip counts.